com.regnosys.rosetta.tests/src/com/regnosys/rosetta/generator/java/function/FunctionGeneratorTest.xtend

```diff
@@ -3571,7 +3620,11 @@ class FunctionGeneratorTest {
 							return ValidationResult.success(NAME, ValidationResult.ValidationType.DATA_RULE,  "Foo", path, DEFINITION);
 						}
 						
-						return ValidationResult.failure(NAME, ValidationResult.ValidationType.DATA_RULE, "Foo", path, DEFINITION, result.getError());
+						String failureMessage = result.getError();
+						if (failureMessage == null) {
+							failureMessage = "Condition " + NAME + " failed.";
+						}
+						return ValidationResult.failure(NAME, ValidationResult.ValidationType.DATA_RULE, "Foo", path, DEFINITION, failureMessage);
```

Comment:
See changes to `DataRuleGenerator`. Made this more robust, as conditions could fail in trivial cases like `condition C: a exists` where `a` is of type `int (0..1)`.